pkg/volume/validation/pv_validation.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/kubernetes/pkg/api"
+)
+
+// MountOptionAnnotation defines mount option annotation used in PVs
+const MountOptionAnnotation = "volume.beta.kubernetes.io/mount-options"
+
+// ValidatePersistentVolume validates PV object for plugin specific validation
+// We can put here validations which are specific to volume types.
+func ValidatePersistentVolume(pv *api.PersistentVolume) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if pv.Spec.GCEPersistentDisk != nil ||
```

Comment:
This is technically correct, but from a maintainability and readability point-of-view not the best approach. It implements the validation of mount options here and `checkMountOption(...)` only serves to generate errors. `ValidatePersistentVolume(...)` should call explicit sub-methods that do both the checking and error generation: so this check should be moved inside `checkMountOption(...)`.